satpy/writers/cf_writer.py

```diff
@@ -499,14 +514,20 @@ def da2cf(dataarray, epoch=EPOCH, flatten_attrs=False, exclude_attrs=None, compr
                 If True, flatten dict-type attributes
             exclude_attrs (list):
                 List of dataset attributes to be excluded
-
+            valid_cf_dataset_name (bool):
+                Prepend dataset name with CHANNEL_ if staring with a digit
         """
         if exclude_attrs is None:
             exclude_attrs = []
 
         new_data = dataarray.copy()
         if 'name' in new_data.attrs:
             name = new_data.attrs.pop('name')
+            if valid_cf_dataset_name and name[0].isdigit():
+                _orig_name = name
+                name = 'CHANNEL_' + name
+                warnings.warn('Rename dataset {} to {}.'.format(_orig_name, name))
+                new_data.attrs['satpy_dataset_name'] = _orig_name
             new_data = new_data.rename(name)
```

Comment:
It would make sense to extract a new method from this, it would the complexity of da2cf lower

Reply:
yes